frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -125,8 +126,12 @@ local function isDocumentComicOrManga(file)
     return supported_filetypes[filetype]
 end
 
-local function isPanelZoomEnabled()
-    return G_reader_settings:nilOrTrue("panel_zoom_allowed")
+local function isPanelZoomEnabled(file)
+    local doc_setting = DocSettings:readSetting("panel_zoom_enabled")
```

Comment:
You can access directly the setting for the currently opened book with:
`self.ui.doc_settings:readSetting("panel_zoom_enabled")`
without re-opening it (and it it has been updated but not yet flushed, you get the updated value).
But as you probably will set the menu to toggle it in this file, you'll probably keep a copy as self.panel_zoom_enabled that you can access directly - and load/save it in the existing:
```lua
function ReaderHighlight:onReadSettings(config)
function ReaderHighlight:onSaveSettings()
```

Reply:
Thanks, saved me quite a bit of time!